### PR TITLE
Support adding Taints on the worker nodes

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR adds support for adding Taints on the worker nodes. Taints are added using the `.providerSpec.Taints` field.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Relevant to #662

**Does this PR introduce a user-facing change?**:
```release-note
Add support for adding Taints on the worker nodes
```

/assign @kron4eg

### CODE REVIEW

File: pkg/apis/kubeone/types.go [hunk 1]
```diff
@@ -167,6 +168,7 @@ type WorkerConfig struct {
 type ProviderSpec struct {
 	CloudProviderSpec   json.RawMessage   `json:"cloudProviderSpec"`
 	Labels              map[string]string `json:"labels"`
+	Taints              []corev1.Taint    `json:"taints"`
```
Comment (xmudrii) [inline pkg/apis/kubeone/types.go]: Should we use `corev1.Taints` directly or have our API field and convert to `corev1.Taints` on runtime?

Comment (kron4eg) [reply pkg/apis/kubeone/types.go]: In machine spec it’s Taints []corev1.Taint `json:"taints,omitempty"`

Comment (xmudrii) [reply pkg/apis/kubeone/types.go]: Done. :+1:

**Comment by xmudrii** [comment]
@kron4eg Fixed, PTAL.

**Pull request merged**